File: src/main/java/no/stunor/origo/eventorapi/model/origo/user/UserRace.java
package no.stunor.origo.eventorapi.model.origo.user;

import java.util.Date;
import java.util.List;
import java.util.Map;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import no.stunor.origo.eventorapi.model.Eventor;
import no.stunor.origo.eventorapi.model.origo.entry.EntryBreak;

@Getter
@Setter
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class UserRace{
    private Eventor eventor;
    private String eventId;
    private String eventName;
    private String raceId;
    private String raceName;
    private boolean canceled;
    private Date raceDate;
    private List<UserCompetitor> userCompetitors;
    private Map<String, Integer> organisationEntries;
    private List<EntryBreak> entryBreaks;
}

File: src/main/java/no/stunor/origo/eventorapi/model/calendar/UserRace.kt
package no.stunor.origo.eventorapi.model.calendar;

import no.stunor.origo.eventorapi.model.Eventor
import java.util.Date;

import no.stunor.origo.eventorapi.model.origo.entry.EntryBreak;
import java.io.Serializable

data class UserRace(
        var eventor: Eventor = Eventor(),
        var eventId: String = "",
        var eventName: String = "",
        var raceId: String = "",
        var raceName: String? = null,
        var canceled: Boolean = false,
        var raceDate: Date? = Date(),
        var userCompetitors: MutableList<UserCompetitor> = mutableListOf(),
        var organisationEntries: MutableMap<String, Int> = mutableMapOf(),
        var entryBreaks: List<EntryBreak> = ArrayList()
) : Serializable